protocols/imap/src/main/java/org/apache/james/imap/processor/AuthenticateProcessor.java

```diff
@@ -140,7 +142,7 @@ private AuthenticationAttempt parseDelegationAttempt(String initialClientRespons
         List<Capability> caps = new ArrayList<>();
         // Only ounce AUTH=PLAIN if the session does allow plain auth or TLS is active.
         // See IMAP-304
-        if (session.isPlainAuthDisallowed()  == false || session.isTLSActive()) {
+        if (session.isPlainAuthEnabled() && (!session.isPlainAuthDisallowed() || session.isTLSActive())) {
```

Comment:
Is this connected to the boolean expression line 63 - 66 above?

Any way we could reuse that?